node/src/proxy_server/mod.rs

```diff
@@ -3635,13 +3667,52 @@ mod tests {
             scheduled_msg,
             &MessageScheduler {
                 scheduled_msg: StreamKeyPurge { stream_key },
-                delay: Duration::from_secs(5)
+                delay: STREAM_KEY_PURGE_DELAY
             }
-        )
+        );
+        TestLogHandler::new()
```

Comment:
The delay-the-purge machinery shouldn't even kick in here, since we're getting notification that the far end is shutting down, rather than sending notification to the far end that the near end is shutting down. That's when we ought to delay the purge. Here, we ought to purge immediately*.

* Ultimately, after the entity beyond the exit Node drops the far end of the connection, we want to wait until the client drops the near end of the connection to purge the StreamKey. That extra bit of life for the StreamKey won't mean anything in HTTP, but it might be important for other protocols. If this is not what we're currently doing, though, that modification is for another card, not this one.